app/main/autoUpdate.ts

```diff
@@ -0,0 +1,143 @@
+import axios, { AxiosRequestConfig } from 'axios';
+import { ipcMain, BrowserWindow } from 'electron';
+import { autoUpdater, UpdateInfo } from 'electron-updater';
+import log from 'electron-log';
+import { createHash, createVerify, Verify } from 'crypto';
+import fs from 'fs';
+
+import {
+    updateAvailable,
+    updateDownloaded,
+    updateVerified,
+    updateError,
+} from '~/constants/ipcRendererCommands.json';
+import {
+    triggerAppUpdate,
+    quitAndInstallUpdate,
+} from '~/constants/ipcCommands.json';
+import { publicKeyUrl } from '~/constants/verification.json';
+
+import { version } from '../package.json';
+import { build } from '../../package.json';
+
+const updateServer = 'https://update.electronjs.org';
+const updateFeed = `${updateServer}/${build.publish.owner}/${build.publish.repo}/${process.platform}-${process.arch}/${version}/`;
+autoUpdater.setFeedURL({
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    ...(build.publish as any),
+    url: updateFeed,
+});
+
+log.transports.file.level = 'info';
+
+autoUpdater.logger = log;
+autoUpdater.autoDownload = false;
+autoUpdater.autoInstallOnAppQuit = false;
+
+interface RealUpdateInfo extends UpdateInfo {
+    downloadedFile: string;
+}
+
+async function getRemoteContent(url: string, binary = false) {
+    const options: AxiosRequestConfig = {};
+
+    if (binary) {
+        options.responseType = 'arraybuffer';
+    }
+
+    const { data } = await axios.get(url, options);
+    return data;
+}
+
+function getSha256Sum(path: string): Promise<string> {
+    const sum = createHash('sha256');
+    const stream = fs.createReadStream(path);
+
+    stream.on('data', (data) => sum.update(data));
+
+    return new Promise((resolve, reject) => {
+        stream.on('end', () => resolve(sum.digest('hex')));
+        stream.on('error', reject);
+    });
+}
+
+function getVerifier(path: string): Promise<Verify> {
+    const verifier = createVerify('RSA-SHA256');
```

Comment:
We should use `ed25519` instead of RSA for making the signature. So you need to change the verification to verify here. Elsewhere we verify signatures using `noble-ed25519` in the desktop wallet, but perhaps crypto can also do it.

Reply:
I adapted the signing and verification functionality to work with keys generated with `ed25519`.